test/jdk/javax/swing/JInternalFrame/InternalFrameBorderTest.java

```diff
@@ -0,0 +1,260 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.AWTException;
+import java.awt.Color;
+import java.awt.GridBagLayout;
+import java.awt.Image;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.image.MultiResolutionImage;
+import java.awt.image.RenderedImage;
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
+import javax.swing.JFrame;
+import javax.swing.JInternalFrame;
+import javax.swing.JLabel;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.imageio.ImageIO;
+
+/*
+ * @test
+ * @bug 8015739
+ * @key headful
+ * @summary Tests whether background color of JInternalFrame is visible
+ * in the border region at different scales by checking the midpoints
+ * and corners of the border.
+ *
+ * @requires (os.family == "windows")
+ * @run main/othervm -Dsun.java2d.uiScale=1 InternalFrameBorderTest
+ * @run main/othervm -Dsun.java2d.uiScale=1.25 InternalFrameBorderTest
+ * @run main/othervm -Dsun.java2d.uiScale=1.5 InternalFrameBorderTest
+ * @run main/othervm -Dsun.java2d.uiScale=1.75 InternalFrameBorderTest
+ * @run main/othervm -Dsun.java2d.uiScale=2 InternalFrameBorderTest
+ * @run main/othervm -Dsun.java2d.uiScale=2.5 InternalFrameBorderTest
+ * @run main/othervm -Dsun.java2d.uiScale=3 InternalFrameBorderTest
+ */
+
+/*
+ * @test
+ * @bug 8015739
+ * @key headful
+ * @summary Tests whether background color of JInternalFrame is visible
+ * in the border region at different scales by checking the midpoints
+ * and corners of the border.
+ *
+ * @requires (os.family == "mac" | os.family == "linux")
+ * @run main/othervm -Dsun.java2d.uiScale=1 InternalFrameBorderTest
+ * @run main/othervm -Dsun.java2d.uiScale=2 InternalFrameBorderTest
+ */
+
+public class InternalFrameBorderTest {
+    private static final int FRAME_SIZE = 300;
+    private static final int INTFRAME_SIZE = 200;
+    private static final int MIDPOINT = INTFRAME_SIZE / 2;
+    private static final int BORDER_THICKNESS = 5;
+
+    private static StringBuffer errorLog = new StringBuffer();
```

Comment:
```suggestion
    private static final StringBuffer errorLog = new StringBuffer();
```
IDE issues a warning that `errorLog` can be final and it should.